src/main/scala/io/phdata/retirementage/loadgen/LoadGenerator.scala

```diff
@@ -27,13 +43,71 @@ object LoadGenerator {
     * @param payloadBytes Payload size
     * @return The test dataframe
     */
-  def generateTable(numRecords: Int, payloadBytes: Int): DataFrame = ???
+  def generateTable(spark: SparkSession, numRecords: Int, payloadBytes: Int): DataFrame = {
+
+    /**
+      * Create the following schema:
+      *   -- id: String (nullable = false)
+      *   -- payload: String (nullable = false)
+      *   -- date: String (nullabe = false)
+      */
+    val schema = StructType(StructField("id", StringType, false) :: Nil)
+      .add(StructField("payload", StringType, false))
+      .add(StructField("date", StringType, false))
+    var dataBuffer = ListBuffer[List[String]]()
+    // Create a string with specified bytes
+    var i          = 0
+    var byteString = ""
+    for (i <- 1 to payloadBytes) {
+      byteString = byteString + "a"
+    }
+    // Create numRecords rows of data with a random UUID and attached a payload of specified bytes and a date
+    var j = 0
```

Comment:
a more functional way of looping x number of times is 
```
(1 to 20).foreach(i => doSomething(i))

or 

for (i <- 1 to 20) {
  doSomething(i)
}
```

using variables `var`, while not especially dangerous here, can lead to a lot of subtle bugs.